tests/test_expression.py

```diff
@@ -1068,6 +1078,23 @@ def test_value_range_simplified() -> None:
         ValueRange(Number(1), Add(Number(21), Number(21))).simplified(),
         ValueRange(Number(1), Number(42)),
     )
+    expr = ValueRange(lower=First("Test"), upper=Sub(Last("Test"), Number(1)))
+
+    def func(expr: Expr) -> Expr:
+        if expr == First("Test"):
+            return Number(1)
+        if expr == Last("Test"):
+            return Number(11)
+        return expr
+
+    assert expr.substituted(func) == ValueRange(lower=Number(1), upper=Sub(Number(11), Number(1)))
+
+    assert_equal(
+        ValueRange(lower=Variable("X"), upper=Variable("Y")).substituted(
+            lambda x: Variable("Z") if isinstance(x, ValueRange) else x
+        ),
+        Variable("Z"),
+    )
```

Comment:
Please have a look at existing code before adding new one and keep it consistent.

This code does not test `simplified`, but `substituted`. Hence it should be in a function called `test_value_range_substituted`.

Reply:
Fixed in https://github.com/Componolit/RecordFlux/pull/381/commits/1983208617f71b6343e34197f95baaae407e107a, I hope is has the correct format now.